### PR TITLE
new meta format; enchanced grouping; minor fixes

### PR DESCRIPTION
**Pull request by Minizbot2012**
New meta format works with all options of TTMP2 format
Enhanced grouping allows individual selection of parts of mod (tested with several mods such as TnF2)
Minor issue: requires reset of mod folder :(

### CODE REVIEW

File: Penumbra/UI/SettingsInterface.cs [hunk 1]
```diff
@@ -598,36 +598,56 @@ private void DrawEditButtons()
...
+                                _plugin.ModManager.CalculateEffectiveFileList();
+                            }
+                            i++;
+                        }
+                        break;
+                    }
+                    case "Single": 
+                    {
+                        var code = conf[g.Key];
+                        if(ImGui.Combo(g.Value.GroupName, ref code, g.Value.Options.Select(x=>x.OptionName).ToArray(), g.Value.Options.ToArray().Length)) {
```
Comment (Ottermandias) [inline Penumbra/UI/SettingsInterface.cs:635]: I think I would skip this if Options has only one entry. Currently Tight & Firm body has a pulldown menu for 'Required Files' where you can't select anything.

Comment (Ottermandias) [reply Penumbra/UI/SettingsInterface.cs:635]: This also implicitly allows for a hidden base option for files that are included in every option. :)

File: Penumbra/Models/GroupInformation.cs [hunk 2]
```diff
@@ -1,189 +1,15 @@
...
-            }
-        }
+    public struct Option {
+        public string OptionName;
+        public string OptionDesc;
+        public Dictionary<string, string> OptionFiles;
+    }
+    public struct InstallerInfo {
+        public string GroupName;
+        public string SelectionType;
```
Comment (NotAdam) [inline Penumbra/Models/GroupInformation.cs:12]: this should be an enum

**Review by NotAdam — COMMENTED** [review]
not really concerned about the file structure in a mod because i'm going to change it anyway

File: Penumbra/Models/GroupInformation.cs (hunk 2 above)
Comment (Minizbot2012) [reply Penumbra/Models/GroupInformation.cs]: changed in a84f2cd

File: Penumbra/UI/SettingsInterface.cs (hunk 1 above)
Comment (Minizbot2012) [reply Penumbra/UI/SettingsInterface.cs]: changed in c7d4cc7

**Comment by NotAdam** [comment]
sorry about the delay - i haven't forgotten, will have a proper look this weekend and merge

**Comment by Minizbot2012** [comment]
No problem, take your time

**Pull request merged**